src/layer/BYOCLayer.ts

```diff
@@ -304,17 +293,17 @@ export class BYOCLayer extends AbstractSentinelHubV3Layer {
         ...getAxiosReqParams(innerReqConfig, CACHE_CONFIG_30MIN),
       };
 
-      const metadataUrl = `${this.getSHServiceRootUrl()}api/v1/metadata/collection/${this.getTypeId()}`;
+      const metadataUrl = `${this.getSHServiceRootUrl()}api/v1/catalog/1.0.0/collections/${this.getTypeId()}`;
       const metadataRes = await axios.get(metadataUrl, commonReqConfig);
-      const metadataBands: BYOCBand[] = metadataRes.data.bands;
-
-      const catalogUrl = `${metadataRes.data.location.catalogUrl}/collections/${this.getTypeId()}`;
-      const catalogRes = await axios.get(catalogUrl, commonReqConfig);
-      const catalogBands: { name: string }[] = catalogRes.data.summaries['eo:bands'];
-
-      // we need bands' sampleTypes that are returned from Metadata API
-      // but want to use only bands that are also returned from Catalog API
-      return metadataBands.filter((mb) => !!catalogBands.find((cb) => cb.name === mb.name));
+      const metadataBands: BYOCBand[] = (
+        metadataRes.data.summaries['eo:bands'] as { name: string }[] | undefined
+      )?.map(({ name }, index) => {
+        return {
+          name,
+          sampleType: metadataRes.data.summaries?.['raster:bands'][index].data_type?.toUpperCase(),
```

Comment:
This assumes that all elements in `eo:bands` have a "sibling" in `raster:bands` at exactly the same index.

I haven't found a collection where this wouldn't be true (PSD collections), but it would be best to check.

Reply:
Great, thanks.